impeller/aiks/image_filter.h

```diff
@@ -275,6 +284,35 @@ class LocalMatrixImageFilter : public ImageFilter {
   std::shared_ptr<ImageFilter> internal_filter_;
 };
 
+/*******************************************************************************
+ ******* RuntimeEffectImageFilter
+ ******************************************************************************/
+
+class RuntimeEffectImageFilter : public ImageFilter {
+ public:
+  RuntimeEffectImageFilter(
+      std::shared_ptr<RuntimeStage> runtime_stage,
+      std::shared_ptr<std::vector<uint8_t>> uniforms,
+      std::vector<RuntimeEffectContents::TextureInput> texture_inputs);
+
+  ~RuntimeEffectImageFilter() = default;
+
+  // |ImageFilter|
+  std::shared_ptr<FilterContents> WrapInput(
+      const FilterInput::Ref& input) const override;
+
+  // |ImageFilter|
+  std::shared_ptr<ImageFilter> Clone() const override;
+
+  // |ImageFilter|
+  void Visit(ImageFilterVisitor& visitor) override { visitor.Visit(*this); }
```

Comment:
OOL this and especially the dtor since it contains vectors and shared pointers.

Reply:
Done